clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -484,6 +502,12 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         }
 #endif
         shard->prev_instr_ = memref;
+        if (next_pc != nullptr) {
+            shard->prev_instr_was_syscall_ = instr_is_syscall(&cur_instr_decoded);
+            shard->prev_instr_decoded_ = &cur_instr_decoded;
+            shard->prev_instr_addr_ = shard->prev_instr_.instr.addr;
+            instr_free(GLOBAL_DCONTEXT, &cur_instr_decoded);
```

Comment:
We do not want instr_free for the current instr here since it is still needed in the next iteration. We want it for the prev instr (before we overwrite it with the cur instr above).

Reply:
Reminder for this unresolved comment.

Reply:
Reminder to do an instr_free for prev_instr_decoded_ (before replacing it with the cur_instr_decoded).

Again, it'll help to reply to each comment explicitly like our code review guide suggests (https://dynamorio.org/page_code_reviews.html#autotoc_md114).

Reply:
Ack. Please see latest patch where I added:
```
            if (shard->prev_instr_decoded_ != nullptr)
                instr_free(GLOBAL_DCONTEXT, shard->prev_instr_decoded_.get());
```